client/scripts/views/components/reaction_button.ts

```diff
@@ -4,66 +4,72 @@ import { default as m, VnodeDOM } from 'mithril';
 import { default as mixpanel } from 'mixpanel-browser';
 
 import app from 'state';
-import { IUniqueId } from 'models';
+import { IUniqueId, Proposal, OffchainComment, OffchainThread, AnyProposal } from 'models';
+import Tooltip from './tooltip';
 
 export enum ReactionType {
   Like = 'like',
   Dislike = 'dislike'
 }
 
 interface IAttrs {
-  proposal: IUniqueId;
+  post: OffchainThread | OffchainComment<any>;
   type: ReactionType;
   displayAsLink?: boolean;
+  tooltip?: boolean;
 }
 
 const ReactionButton: m.Component<IAttrs> = {
   view: (vnode: m.VnodeDOM<IAttrs>) => {
-    const proposal = vnode.attrs.proposal;
-    const type : string = vnode.attrs.type;
-    const reactions = app.reactions.getByProposal(proposal);
-
+    const { post, type, displayAsLink, tooltip } = vnode.attrs;
+    const reactions = app.reactions.getByPost(post);
+    const a = app;
     let dislikes;
     let likes;
     if (type === ReactionType.Like) likes = reactions.filter((r) => r.reaction === 'like');
     if (type === ReactionType.Dislike) dislikes = reactions.filter((r) => r.reaction === 'dislike');
 
     const disabled = !app.vm.activeAccount;
-
-    const rxn = reactions.find((r) =>
-      r.reaction && app.vm.activeAccount && r.author === app.vm.activeAccount.address);
+    const activeAddress = app.vm.activeAccount?.address;
+    const rxn = reactions.find((r) => r.reaction && r.author === activeAddress);
     const hasReacted : boolean = !!rxn;
     let hasReactedType;
     if (hasReacted) hasReactedType = rxn.reaction;
 
-    return m('.ReactionButton', {
-      class: (disabled ? 'disabled' : type === hasReactedType ? 'active' : '') +
-        (vnode.attrs.displayAsLink ? ' as-link' : ''),
+    const reactors = (likes || dislikes).map((rxn_) => {
+      const reactor = app.profiles.getProfile(app.activeChainId(), rxn_.author);
+      return m('.reacting-user', reactor.displayName);
+    });
```

Comment:
Seems like we should truncate this to something like 10? and then if the list of reactors is longer than this, add in an additional row that says (and X other reactions)

Reply:
Yup, great catch. I'll slice it at 10.